packages/interactivity/src/init.ts

```diff
@@ -29,6 +29,38 @@ export const initialVdom = new WeakMap< Element, ComponentChild[] >();
 
 // Initialize the router with the initial DOM.
 export const init = async () => {
+	const pendingNodes = new Set();
+
+	const intersectionObserver = new window.IntersectionObserver(
+		async ( entries ) => {
+			for ( const entry of entries ) {
+				if ( ! entry.isIntersecting ) {
+					continue;
+				}
+
+				const node = entry.target;
+				intersectionObserver.unobserve( node );
+				pendingNodes.delete( node );
+				if ( pendingNodes.size === 0 ) {
+					intersectionObserver.disconnect();
+				}
+
+				if ( ! hydratedIslands.has( node ) ) {
+					const fragment = getRegionRootFragment( node );
+					const vdom = toVdom( node );
```

Comment:
In the old logic, this used to populate `initialVdom`. Is this no longer needed? If not, is there even any value in keeping the `initialVdom` const around?

Reply:
It seems this is a merge conflict resolution on my part. This may be needed:

```diff
--- a/packages/interactivity/src/init.ts
+++ b/packages/interactivity/src/init.ts
@@ -48,6 +48,7 @@ export const init = async () => {
 				if ( ! hydratedIslands.has( node ) ) {
 					const fragment = getRegionRootFragment( node );
 					const vdom = toVdom( node );
+					initialVdom.set( node, vdom );
 					await splitTask();
 					hydrate( vdom, fragment );
 					await splitTask();
```

Reply:
Looks like it is only being used here:

https://github.com/WordPress/gutenberg/blob/ef7afefc5988eaab30da0025aa48bfbddb09a79f/packages/interactivity-router/src/index.ts#L189-L200

Reply:
Introduced in https://github.com/WordPress/gutenberg/pull/58496

Reply:
Restored `initialVdom.set( node, vdom )` in 387ab6cb46c378e7e4499357c7f9fafc50ea1f9d.

I found that this code in `interactivity-router` runs as the page loads when the "iAPI: full page client side navigation" experiment is enabled, _and_ it also runs when interacting with a block that makes use of client-side navigation (i.e. the Query block when "Reload full page" is disabled). Nevertheless, the behavior of the page seems to work just as well whether or not `initialVdom.set( node, vdom )` is added here.

@DAreRodz For this page cache, is it a problem that `initialVdom` is not initially populated with all of the interactive regions of the page since they get added only when they come into view?